### ISSUE TITLE
Weird colors in success notice after animation

### ISSUE THREAD

**Issue by nitriques** [symphonycms/symphony-2#2466]
Is this a bug ? Or as designed ?

![image](https://cloud.githubusercontent.com/assets/771169/8579992/0044f056-2586-11e5-870a-73fa4be971e8.png)

I would rather see this:

![image](https://cloud.githubusercontent.com/assets/771169/8580002/0d9052f0-2586-11e5-9757-a46a27aa835e.png)

cc @nilshoerrmann

**Comment by michael-e** [comment]
Buggg.

**Comment by nitriques** [comment]
;) Let's wait for the designer's comment hahaha

**Comment by jurajkapsz** [comment]
I also noticed this in webkit browsers.

**Comment by nilshoerrmann** [comment]
That's by design of course! You should never be forced to create another entry ;)
No, seems like an old Webkit bug is popping up again when transitioning from dark to light colours …

**Comment by nitriques** [comment]
:) Ok thanks Nils I'll try to fix this up without breaking anything else ;)

**Comment by nitriques** [comment]
I am so a lame designer (even, not a graphical designer at all), I really wanted to check with you first :P

**Comment by jurajkapsz** [comment]
Maybe a way to do it is through js.

**Comment by nitriques** [comment]
> Maybe a way to do it is through js.

? Everything is done in css right now... Should be pretty easy to fix.

**Comment by nitriques** [comment]
Finally @jurajkapsz was right: we must do it in js... this is the bug: http://stackoverflow.com/questions/8490828/css3-color-transition-not-working-in-chrome

:visited links can't be animated...

**Comment by jurajkapsz** [comment]
yeah, js will be a sure thing.

**Comment by nitriques** [comment]
Adding a cache buster (?cache=<timestamp>) would solve the problem. But it ain't pretty.

**Comment by brendo** [comment]
> Is this a bug ? Or as designed ?

Bug.

> Adding a cache buster (?cache=) would solve the problem. But it ain't pretty.

Agreed, I don't like this. 

Is another solution to use JS to add classes? I notice the current implementation is quite elegant in that it uses an infinite CSS transition, but if it's not working then it's not totally useful to us :) Perhaps we could listen to the `transitionend` event and add the appropriate class for better compatibility?

**Comment by nitriques** [comment]
> transitionend 

Good idea. But still, the css animation won't work.

**Comment by johnpuddephatt** [comment]
It looks like transitions work, even though animations don't. [Codepen demonstration](http://codepen.io/anon/pen/MwBvqr) – can anyone else confirm this is working in Chrome?

Would it be acceptable to rely on the .js-active class on the html element to trigger this?

If so, the following could work:

````
.js-active .notice.success a {
  color:#bdc3c3;
  border-color:#7c7c7c;
  transition: .5s ease-out 10s;
}
````

**Comment by nitriques** [comment]
Sadly, it does not works. The link always stay red. :(

**Comment by johnpuddephatt** [comment]
How strange. Working for me on Chrome.

Animation doesn't, but transition does...

**Comment by nitriques** [comment]
I am on windows right now... Latest version. Always red before and after clicks.

**Comment by johnpuddephatt** [comment]
Ok. I'm on OS X. The inconsistency is perhaps not too surprising given that it's a bug...

**Comment by johnpuddephatt** [comment]
Oh, hang on. If it stays red that's maybe a specificity issue, not an animation issue? Does adding !important to the .visited color declaration help?

**Comment by nitriques** [comment]
Indeed.

**Comment by nitriques** [comment]
Same result :(

**Comment by johnpuddephatt** [comment]
Two thoughts:

– you are waiting a few seconds after clicking, right?
– what happens if you delete the transition?

**Comment by nitriques** [comment]
> you are waiting a few seconds after clicking, right?

No I was not! It DOES works! Yeah!!!

**Comment by johnpuddephatt** [comment]
:scream_cat:

OK, that's good! Seems like a relatively non-hacky solution. Does rely on javascript, but that's where the discussion was heading anyway, and it doesn't rely on writing any new javascript which seems convenient.

**Comment by nitriques** [comment]
Yup, I'll patch this up right now!

**Comment by johnpuddephatt** [comment]
Assuming .js-active is being added on DOM ready (or after) it should work!

**Comment by brendo** [comment]
Hey @nitriques, how'd the patch go? Wouldn't mind collating together a couple of fixes and pushing through 2.6.3 :)

**Comment by nitriques** [comment]
Yes! As you know, I've been quite busy this week. Might have the time to properly test it on Monday.

**Comment by brendo** [comment]
Bump on this one, it's the only thing stopping a 2.6.3 from being shipped :)

**Comment by nitriques** [comment]
Oh shit, let me fix this right now.

**Comment by nitriques** [comment]
There you go.

@johnpuddephatt  I did not used `.js-active` since those classes (`.js-*`) should not be in the css files.

**Comment by brendo** [comment]
I can confirm this fixes the behaviour in this ticket, but can anyone also confirm that in Chrome all the hover styles are totally ignored for notices? If I 'disable' the transition, hover effects work a treat, but with the transition it's like Chrome totally ignores hover!

**Comment by johnpuddephatt** [comment]
Yep I can confirm seeing the same problems with hover.

Moving the transition delay from CSS into Javascript seems to sort it – as @nitriques fix relied on using setTimeout in JS I couldn't see any harm in doing this.

I've also added in a hover style for the anchor when it's dimmed as this seemed to be missing.

I've only tested this in modern Chrome/FF/Safari, but it seems to be working as expected, be good if someone can confirm!

**Comment by nitriques** [comment]
>  but can anyone also confirm that in Chrome all the hover styles are totally ignored for notices?

Damn...

> I've also added in a hover style for the anchor when it's dimmed as this seemed to be missing.

Could you please share this code with us :)

**Comment by nitriques** [comment]
This can be closed following the 2.6.3 release.

**Comment by brendo** [comment]
Hmm, that PR doesn't seem to have any effect in Chrome. Surprising how strange this is. I can get very weird and mixed results in Chrome by hover over the link during the transition and then waiting for the dimmed state to be added.

I've assembled our 'desired' (most elegant) setup with a [Codepen](http://codepen.io/anon/pen/WvWBaW). For me, Firefox is the only browser that performs correctly. Webkit seems to fail in that once that transition is done, no other states will work.

Even with the current patch, and the proposed #2479, the behaviour is still inconsistent and hover/focus states are cobbled. What a curious bug.

**Comment by johnpuddephatt** [comment]
Yes is it pretty strange.

My PR was based on [this Codepen](http://codepen.io/anon/pen/zGXVLp) which is working for me in Chrome/FF – let me know if that's working for you.

**Comment by johnpuddephatt** [comment]
@brendo did you check that Codepen?

I've also just realised that adding `animation: none` to the :hover state in your example seems to do the trick. [See here](http://codepen.io/johnpuddephatt/pen/mJYvbr).

**Comment by brendo** [comment]
Hey @johnpuddephatt. Yep! I tried the Codepen, still doesn't work correctly in Chrome. The `animation: none` allows the hover effect to work, but when you hover away from the link, the animation restarts. So you'll notice the link goes back to white, and then 4s later it becomes blue again.

It should be:

1. Link is white for 4s (during this time `:hover` should make the text yellow)
2. Link fades to blue at the same time the background fades
3. Hovering over the _blue_ link should change the text colour to yellow
4. On hover out/mouse away from the link, the text should remain blue

The crux seems to be once the animation has occurred, Blink/Webkit cannot override the link colour with hover. Firefox can though.

**Comment by johnpuddephatt** [comment]
Oh yes, the animation is restarted :/

What problem(s) are you having with the JS triggered one <http://codepen.io/anon/pen/zGXVLp>? That’s working for me in Chrome + OS X… hover states working before and after the alert is ‘dimmed’.

**Comment by brendo** [comment]
Ah ha, yep can confirm that pen works!

**Comment by nitriques** [comment]
So have we found a fix ? :D

**Comment by brendo** [comment]
@johnpuddephatt Is this fix the same that's in #2479?

**Comment by johnpuddephatt** [comment]
Should be yes!

**Comment by brendo** [comment]
This is now merged into `2.6.x`
